### ISSUE TITLE
failure to hide team tabs from context menu

### ISSUE THREAD

**Issue by caver456** [ncssar/radiolog#406]
right clicking the tab 'SAR5' at the search (or was it 'Sar5'?  either way there was no space) then clicking Hide Tab had no effect, and no errors in the log.

Duplicated this at home, tab name was 'Sar5'.

Looks like it's a case mismatch issue.  The entry was called as 'SAR 5' by typing hotkeys 's' then '5' resulting in tab 'Sar5': 

```
062048:newEntry called with these values:
062048:['0620', 'FROM', 'SAR 5', 'asdf', None, '', 1569417645.0179813, None, None, None]
062048:new team: newTeamName=SAR 5 extTeamName=z_SAR00005 niceTeamName=Sar 5 shortNiceTeamName=Sar5
062048:extTeamNameList before sort:['z_Team00001', 'z_Team00002', 'z_Team00003', 'spacer1', 'z_SAR00005']
062048:grouped tabs:{'NCSOcmd': [], 'NCSO': [], 'NCPD': [], 'TPD': [], 'GVPD': [], 'CHP': [], 'Numbers': ['z_Team00001', 'z_Team00002', 'z_Team00003'], 'other': ['z_SAR00005']}
062048:extTeamNameList after sort:['z_Team00001', 'z_Team00002', 'z_Team00003', 'spacer1', 'z_SAR00005']
```

then the hide-tab attempt:

```
062058:tab context menu requested: pos=PyQt5.QtCore.QPoint(197, 22)
062058:  i=5  tabRect=PyQt5.QtCore.QPoint(180, 34):PyQt5.QtCore.QPoint(232, 0)
062058:  extTeamName=z_SAR00005  niceTeamName=Sar 5
062059:deleting team tab: teamName=Sar 5  extTeamName=z_Sar00005  niceTeamName=Sar 5
062059:  teamNameList before deletion:['Team 1', 'Team 2', 'Team 3', 'Team 6', 'dummy', 'Sar 5', 'Sar 6']
062059:  extTeamNameList before deletion:['z_Team00001', 'z_Team00002', 'z_Team00003', 'z_Team00006', 'spacer1', 'z_SAR00005', 'z_SAR00006']
062059:  extTeamNameList after delete: ['z_Team00001', 'z_Team00002', 'z_Team00003', 'z_Team00006', 'spacer1', 'z_SAR00005', 'z_SAR00006']
```

and a clean hide-tab for a team# for comparison:

```
062115:tab context menu requested: pos=PyQt5.QtCore.QPoint(96, 29)
062115:  i=2  tabRect=PyQt5.QtCore.QPoint(80, 34):PyQt5.QtCore.QPoint(119, 0)
062115:  extTeamName=z_Team00003  niceTeamName=Team 3
062116:deleting team tab: teamName=Team 3  extTeamName=z_Team00003  niceTeamName=Team 3
062116:  teamNameList before deletion:['Team 1', 'Team 2', 'Team 3', 'Team 6', 'dummy', 'Sar 5', 'Sar 6']
062116:  extTeamNameList before deletion:['z_Team00001', 'z_Team00002', 'z_Team00003', 'z_Team00006', 'spacer1', 'z_SAR00005', 'z_SAR00006']
062116:  i=2
062116:  deleted proxyModelList index 2
062116:Freeing hotkey '3' which was used for callsign 'Team 3'
062116:checking tab#1:Team 2
062116:checking tab#2:Team 6
062116:checking tab#3:Spacer 1
062116:checking tab#4:Sar 5
062116:checking tab#5:Sar 6
062116:  extTeamNameList after delete: ['z_Team00001', 'z_Team00002', 'z_Team00006', 'spacer1', 'z_SAR00005', 'z_SAR00006']
```

**Comment by caver456** [comment]
Not sure if this is the same root cause, but, team tabs failed to hide several times during a large campaign search recently.  Got several photos of the log files similar to above, at https://photos.app.goo.gl/ZCvr8SMJV6A8cBiq8

One of the more straightforward ones is here:
![image](https://user-images.githubusercontent.com/18752102/185792711-a7b0979e-0363-449e-8dd9-f5fb3362f955.png)

Why is 'team00206' in extTeamNameList rather than z_Team00206?  May need to look earlier in the log file (radiolog_log_2022_08_08_080430.txt) to see what happened when that team was added.

Also, this example of case sensitivity:
![image](https://user-images.githubusercontent.com/18752102/185792862-6c089e11-c1be-4151-9bbb-65806314cce7.png)

Note that on the tab context menu request, extTeamName is uppercase, but on the next line, extTeamName is lowercase causing the failure to match in extTeamNameList.

Anyway - need to account for all of the cases in those photos before closing this issue.

**Comment by caver456** [comment]
Before applying the fix, was able to reproduce the error with a minimal test case:

```
064834:openNewEntry called:key=s callsign=None formattedLocString=None fleet=None dev=None origLocString=None amendFlag=False amendRow=None
064838:Accepted
064838:vText:
064838:newEntry called with these values:
064838:['0648', 'FROM', 'SAR 5', 'STARTING ASSIGNMENT', None, 'Working', 1663768114.8882046, None, None, None]
064838:new team: newTeamName=SAR 5 extTeamName=z_SAR00005 niceTeamName=Sar 5 shortNiceTeamName=Sar5
064838:extTeamNameList before sort:['dummy', 'z_SAR00005']
064838:grouped tabs:{'Numbers': [], 'other': ['dummy', 'z_SAR00005']}
064838:extTeamNameList after sort:['z_SAR00005']
064838:next available hotkey:1
064838:Accepted2
064838:  writing C:\Users\caver\Documents\New_Incident_2022_09_21_064834.csv
064838:  done writing C:\Users\caver\Documents\New_Incident_2022_09_21_064834.csv
064838:  writing C:\Users\caver\Documents\New_Incident_2022_09_21_064834_clueLog.csv
064838:  done writing C:\Users\caver\Documents\New_Incident_2022_09_21_064834_clueLog.csv
064841:tab context menu requested: pos=PyQt5.QtCore.QPoint(23, 21)
064841:  i=0  tabRect=PyQt5.QtCore.QPoint(0, 31):PyQt5.QtCore.QPoint(52, 0)
064841:  extTeamName=z_SAR00005  niceTeamName=Sar 5
064842:deleting team tab: teamName=Sar 5  extTeamName=z_Sar00005  niceTeamName=Sar 5
064842:  teamNameList before deletion:['Sar 5', 'dummy']
064842:  extTeamNameList before deletion:['z_SAR00005']
064842:  extTeamNameList after delete: ['z_SAR00005']
```

**Comment by caver456** [comment]
Adding the same code as in the fix to #393 just before the comparison:

		# 406: apply the same fix as #393:
		# if the new entry's extTeamName is a case-insensitive match for an
		#   existing extTeamName, use that already-existing extTeamName instead
		for existingExtTeamName in self.extTeamNameList:
			if extTeamName.lower()==existingExtTeamName.lower():
				extTeamName=existingExtTeamName
				break

After applying the fix - works as expected - though it's an important note that this if #393 wasn't already fixed, we would have the potential for multiple tabs closing at the same time:

```
065142:openNewEntry called:key=s callsign=None formattedLocString=None fleet=None dev=None origLocString=None amendFlag=False amendRow=None
065147:Accepted
065147:vText:
065147:newEntry called with these values:
065147:['0651', 'FROM', 'SAR 5', 'COMPLETED ASSIGNMENT', None, '', 1663768302.9285388, None, None, None]
065147:new team: newTeamName=SAR 5 extTeamName=z_SAR00005 niceTeamName=Sar 5 shortNiceTeamName=Sar5
065147:extTeamNameList before sort:['dummy', 'z_SAR00005']
065147:grouped tabs:{'Numbers': [], 'other': ['dummy', 'z_SAR00005']}
065147:extTeamNameList after sort:['z_SAR00005']
065147:next available hotkey:1
065147:Accepted2
065147:  writing C:\Users\caver\Documents\New_Incident_2022_09_21_065141.csv
065147:  done writing C:\Users\caver\Documents\New_Incident_2022_09_21_065141.csv
065147:  writing C:\Users\caver\Documents\New_Incident_2022_09_21_065141_clueLog.csv
065147:  done writing C:\Users\caver\Documents\New_Incident_2022_09_21_065141_clueLog.csv
065149:tab context menu requested: pos=PyQt5.QtCore.QPoint(29, 22)
065149:  i=0  tabRect=PyQt5.QtCore.QPoint(0, 31):PyQt5.QtCore.QPoint(52, 0)
065149:  extTeamName=z_SAR00005  niceTeamName=Sar 5
065150:deleting team tab: teamName=Sar 5  extTeamName=z_SAR00005  niceTeamName=Sar 5
065150:  teamNameList before deletion:['Sar 5', 'dummy']
065150:  extTeamNameList before deletion:['z_SAR00005']
065150:  i=0
065150:  deleted proxyModelList index 0
065150:Freeing hotkey '1' which was used for callsign 'Sar 5'
065150:  extTeamNameList after delete: []
```
